compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/tree/types/BLangArrayType.java

```diff
@@ -39,6 +39,11 @@
     public BLangType elemtype;
     public List<BLangExpression> sizes = new ArrayList<>();
 
+    public int inferredArrayValidateState = 0;  // (-1) - Invalid usage (not in correct context)
```

Comment:
Any specific reason to use integer here? cant we use a enum with these values?

Reply:
Will use enum.